From c4c0ba932bfd3b56910503030fea55e6ef0f91f4 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 16:14:29 -0400
Subject: [ethernet] be2net: Set skb->csum_level for encapsulated checksum

Message-id: <1410192881-30702-67-git-send-email-ivecera@redhat.com>
Patchwork-id: 92465
O-Subject: [RHEL7.1 PATCH 66/78] benet: Set skb->csum_level for encapsulated checksum
Bugzilla: 1088575
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>

BZ: #1088575

Upstream commit(s):
commit b6c0e89dd3a1b22e77fb495a0364a6883cfe81be
Author: Tom Herbert <therbert@google.com>
Date:   Wed Aug 27 21:27:17 2014 -0700

    benet: Set skb->csum_level for encapsulated checksum

    Set skb->csum_level instead of skb->encapsulation when indicating
    CHECKSUM_UNNECESSARY for an encapsulated checksum.

    Signed-off-by: Tom Herbert <therbert@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index ee03df8..aafdd96 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -1679,7 +1679,7 @@ static void be_rx_compl_process(struct be_rx_obj *rxo, struct napi_struct *napi,
  if (netdev->features & NETIF_F_RXHASH)
   skb_set_hash(skb, rxcp->rss_hash, PKT_HASH_TYPE_L3);
 
- skb->encapsulation = rxcp->tunneled;
+ skb->csum_level = rxcp->tunneled;
  skb_mark_napi_id(skb, napi);
 
  if (rxcp->vlanf)
@@ -1737,7 +1737,7 @@ static void be_rx_compl_process_gro(struct be_rx_obj *rxo,
  if (adapter->netdev->features & NETIF_F_RXHASH)
   skb_set_hash(skb, rxcp->rss_hash, PKT_HASH_TYPE_L3);
 
- skb->encapsulation = rxcp->tunneled;
+ skb->csum_level = rxcp->tunneled;
  skb_mark_napi_id(skb, napi);
 
  if (rxcp->vlanf)
-- 
1.7.1